datalad/support/annexrepo.py

```diff
@@ -1604,12 +1608,29 @@ def _get_to_be_added_recs(paths):
         else:
             if backend:
                 options.extend(('--backend', backend))
-            for r in self._call_annex_records(
+            yield from self._maybe_batched_add(
+                files, options, jobs, expected_additions, batch=None)
+
+    def _maybe_batched_add(self, files, options, jobs, expected_additions, batch=None):
+        # Common code to be reused across existing implementations of "annex add" invocation
+        if batch or (batch is None and len(files) > 1):
```

Comment:
This means that the else-path is executed if, for example, `b == ""` and `len(files) > 1`. I guess that is intended?

Reply:
well, `batch` should be a bool, so `""` would be taken as `False` and thus going into `else` since "explicitly" instructed to not batch. So IMHO that was ok.